cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -179,6 +182,7 @@ public void deleteSchedule(Id.Program program, SchedulableProgramType programTyp
       scheduler.deleteSchedule(program, programType, scheduleName);
     } catch (NotFoundException e) {
       LOG.trace("Could not delete schedule", e);
+      Throwables.propagate(e);
```

Comment:
Because the following method doesn't throw a ``NotFoundException``. In the following method (deleting all schedules for one program), we have no way of knowing if the program doesn't exist, or if there is no schedule for an existing program, hence we can't throw that exception.